module/scmi_clock/src/mod_scmi_clock.c

```diff
@@ -610,11 +642,11 @@ FWK_WEAK int mod_scmi_clock_config_set_policy(
          *
          * This is the last agent to set the clock STOPPED.
          */
-        if (clock_count[clock_dev_id] != 1) {
+        if (clock_count[clock_idx] != 1) {
             status = FWK_E_STATE;
         }
         if (policy_commit == MOD_SCMI_CLOCK_POST_MESSAGE_HANDLER) {
-            clock_count[clock_dev_id]--;
+            clock_count[clock_idx]--;
         }
```

Comment:
The decrement should be done only in post message handler otherwise you will never be able to decrement clock_count[clock_idx] because if clock_count[clock_idx] != 1 you don't do post the message  so once clock_count[clock_idx] > 1 you will never decrement it